test/firebase_auth_mocks_test.dart

```diff
@@ -768,13 +771,13 @@ void main() {
       signedIn: true,
     );
     final decodedToken =
-        JwtDecoder.decode(await auth.currentUser!.getIdToken());
+        JwtDecoder.decode(await auth.currentUser!.getIdToken() as String);
```

Comment:
Instead of using `as String`, would you mind rewriting this as:
```dart
JwtDecoder.decode((await auth.currentUser!.getIdToken())!);
```
This way we show that the type is right, and we're only forcing it to be non-nullable.